javascript/atoms/action.js

```diff
@@ -75,27 +93,41 @@ bot.action.checkInteractable_ = function(element) {
 
 
 /**
- * Clears the given {@code element} if it is a editable text field.
+ * Clears the given {@code element} if it is a editable or selectable field.
  *
  * @param {!Element} element The element to clear.
- * @throws {bot.Error} If the element is not an editable text field.
+ * @throws {bot.Error} If the element is not an editable or selectable field.
  */
 bot.action.clear = function(element) {
   bot.action.checkInteractable_(element);
   if (!bot.dom.isEditable(element)) {
-    throw new bot.Error(bot.ErrorCode.INVALID_ELEMENT_STATE,
-        'Element must be user-editable in order to clear it.');
+    if (!bot.dom.isSelectable(element)) {
+      throw new bot.Error(bot.ErrorCode.INVALID_ELEMENT_STATE,
+          'Element must be user-editable in order to clear it.');
+    }
   }
 
   bot.action.LegacyDevice_.focusOnElement(element);
+  if (bot.dom.isSelectable(element)) {
+    // If radio or checkbox is checked, then set the checkedness to false.
+    // https://w3c.github.io/webdriver/webdriver-spec.html#element-clear
+    if (bot.dom.isSelected(element)) {
+      element.checked = false;
+      bot.events.fire(element, bot.events.EventType.CHANGE);
+    }
+  }
+
   if (element.value) {
     element.value = '';
     bot.events.fire(element, bot.events.EventType.CHANGE);
-  } else if (bot.dom.isElement(element, goog.dom.TagName.INPUT) &&
-             (element.getAttribute('type') && element.getAttribute('type').toLowerCase() == "number")) {
-    // number input fields that have invalid inputs
-    // report their value as empty string with no way to tell if there is a
-    // current value or not
+  } else if (bot.dom.isEditable(element) && !bot.dom.isTextual(element)) {
```

Comment:
why wouldn't we call bot.dom.isNonTextual (and not have to negate the value).

Also it seems like the type="number" case wouldn't evaluate to true here? (since according to bot.dom it is textual)